html/modules/custom/reliefweb_subscriptions/src/Controller/UnsubscribeForm.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Drupal\reliefweb_subscriptions\Controller;
+
+/**
+ * @file
+ * Unsubscribe controller.
+ */
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * A Bookmarks toggle controller.
+ */
+class UnsubscribeForm extends ControllerBase {
+  /**
+   * Account.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $account;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(AccountInterface $account) {
+    $this->account = $account;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * Add or remove node from bookmarks.
+   */
+  public function unsubscribe(AccountInterface $user) {
+    if ($this->account->isAnonymous()) {
+      return $this->redirect('user.login', [], [
```

Comment:
This differs from the prod behavior which tries to follow some best practices and exposes the list of the user's subscriptions so that they can unsubscribe without having to log in. It would be good to restore this functionality.